### DISCUSSION TITLE
关于class property decorator支持的配置问题

### DISCUSSION THREAD

**Discussion by ahuigo** [umijs/umi#10945] [Q&A]
写了一个关于class property decorator的demo:

相关ts配置已经开启，但是不知为何，运行时没有效果。

复现用例: https://github.com/ahuigo/umi-demo/blob/c837ef56807c5b5e27ee4725b49929a27c597602/src/pages/decorator.tsx#L27

```
{
  "extends": "./src/.umi/tsconfig.json",
  "compilerOptions": {
    "strictPropertyInitialization": true,
    "skipLibCheck": true,
    "strict": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true
  }
}
```
我还缺少什么配置吗？

**Comment by fz6m** [comment]
装饰器使用方法有问题，可以看这篇文章的评论区：

https://dev.to/danywalls/using-property-decorators-in-typescript-with-a-real-example-44e

你可以试试 `User.prototype.password = '2'`

**Reply by ahuigo** [reply]
看过这篇。应该是es2022要把关闭useDefineForClassFields才能用prototype。

只是我在umijs 中加了`    "useDefineForClassFields": false` 看起来没生效

https://www.typescriptlang.org/play?experimentalDecorators=true&emitDecoratorMetadata=true&target=9&useDefineForClassFields=false#code/GYVwdgxgLglg9mABAWRmAFAGxgWxlALkTBBwCMBTAJwEpEBvAWAChFEqKoQqlRJYEidFACGVAOaciAeTIArCtAA0iAA5U4q6lACeAaQo6iAZyhU04uk1ZtEmTogBuIzCAomzFgNwtbiCAimiJJQUNSIALyIfNDwSOhWvn7snNxIzq4UPjZsAL7ZfgFgQcacYVSR0eCxguhgFADuAGouHuZglgxJhYFw9gB0mHDi6ABEpaHUoyr1zS40BckwwEKzLZiDFB1QABaIADx2uPiJOclsohKc-dQaVMZRAAYAmnDcaiLGxg1wVAAmiGMOzemABlEQZBg4kkFV2IiQABJ6Ng8FBckokWsXLlHt1bLlEBRMKUumc-Bk3JUsZhFn5cnj8njZApoP0-hRgGgKAAFDRaKi6YRiEIqdSabT6QwqaznEJEELlJR4tgTExlagMhZJRnMenMFgQTCfYyIACqpQqMsQIAtYBEOHcgM8HUWAAFUBgABw0JKqY0-f5tbxJIqmKggaC-dA26h2h1Bjqi-2-P4JzpWi47GDGfoxnj2iiVPNxrJ43bZ-p+r4BgFRKvfFOLPV6g2BKCIP7wnTm8JRWZmi1jACC9gAHtNEKMAIwAJgAzAAWUZa5idsDdi2V5P-SqjBcAVgAbAB2UbZUN9CiDYboADka57VAIt6Ua431BXF4GQxGt8fL4HahKw0KA4F0LQfX1ZggA
